lib/Sema/CSSimplify.cpp

```diff
@@ -1561,6 +1561,7 @@ ConstraintSystem::matchTypesBindTypeVar(
   }
 
   assignFixedType(typeVar, type);
+  invalidateBindings(typeVar);
```

Comment:
Are there any other callers of assignFixedType() that need to do this?

Reply:
I am considering unifying both assign and merge paths, want to run perf test first before I spend more time on this...